<!--
Auto-generated by: https://github.com/threlte/threlte/tree/main/packages/gltf
Command: npx @threlte/gltf@1.0.0-next.13 C:\Users\Aaron\Documents\FunBit\static\models\Rock_2.gltf --root /models/ --types --printwidth 120 --precision 2
-->

<script lang="ts">
	import type * as THREE from 'three';
	import { Group } from 'three';
	import { T, type Props, type Events, type Slots, forwardEventHandlers } from '@threlte/core';
	import { useGltf, useTexture } from '@threlte/extras';

	type $$Props = Props<THREE.Group>;
	type $$Events = Events<THREE.Group>;
	type $$Slots = Slots<THREE.Group> & { fallback: {}; error: { error: any } };

	export const ref = new Group();

	type GLTFResult = {
		nodes: {
			Rock_2: THREE.Mesh;
		};
		materials: {
			Rock: THREE.MeshStandardMaterial;
		};
	};

	const gltf = useGltf<GLTFResult>('/Ultimate-Stylized-Nature/Rock_2.gltf');
	const texture1 = useTexture('/Ultimate-Stylized-Nature/Textures/Rocks.png');

	const assets = Promise.all([gltf, texture1]);

	const component = forwardEventHandlers();
</script>

<T is={ref} dispose={false} {...$$restProps} bind:this={$component}>
	{#await assets}
		<slot name="fallback" />
	{:then [gltf, t1]}
		<T.Mesh geometry={gltf.nodes.Rock_2.geometry}>
			<T.MeshStandardMaterial map={t1} map.offset={[0, 0.4]} />
		</T.Mesh>
	{:catch error}
		<slot name="error" {error} />
	{/await}

	<slot {ref} />
</T>
